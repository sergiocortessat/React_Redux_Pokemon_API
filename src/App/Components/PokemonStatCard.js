import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import '../../Styles/PokemonStatCard.scss';
import ProgressBar from './ProgressBar';

const PokemonStatCard = ({ pokemon }) => {
  const [completed, setCompleted] = useState([]);

  const pokemonStats = [
    { name: 'HP', stat: pokemon.HP, color: '#00ff00' },
    { name: 'ATTACK', stat: pokemon.Attack, color: '#ff0000' },
    { name: 'DEFENSE', stat: pokemon.Defense, color: '#0000ff' },
    { name: 'SPECIAL ATTACK', stat: pokemon.SpecialAttack, color: '#ffff00' },
    { name: 'SPECIAL DEFENSE', stat: pokemon.SpecialDefense, color: '#00ffff' },
    { name: 'SPEED', stat: pokemon.Speed, color: '#ff00ff' },
  ];

  useEffect(() => {
    setCompleted(([...pokemonStats]));
  }, [pokemon]);

  return (
    <>
      <div className="stat-name">
        <h1>
          {`#${pokemon.id}`}
          {' - '}
          {pokemon.english}
        </h1>
      </div>
      <div className="wrapper">
        <div className="inner">
          <div className="info">
            <h2>
              ID
              {' '}
              #
              {pokemon.id}

            </h2>
            <h2>
              HEIGHT
              {' '}
              {pokemon.height}
            </h2>
            <h2>
              WEIGHT
              {' '}
              {pokemon.weight}
            </h2>
            {pokemon.type && (
            <div className="stat-type">
              <h2>TYPE</h2>
              <h3 className="type0">
                {pokemon.type[0]}
              </h3>
              <h3 className="type1">
                {pokemon.type[1]}
              </h3>
            </div>
            )}
            {pokemon.evolution && (
            <h2>
              NEXT-EVOLUTION
              {' '}
              {pokemon.evolution.next ? pokemon.evolution.next[0][1].toUpperCase() : 'NO-EVOLUTION'}
            </h2>
            )}
          </div>
          {pokemon.ability && (
          <div className="stat-ability">
            <h2>ABILITY</h2>
            <h3 className="ability1">
              {pokemon.ability[0] ? pokemon.ability[0][0] : '--'}
            </h3>
            <h3 className="ability2">
              {pokemon.ability[1] ? pokemon.ability[1][0] : '--'}
            </h3>
          </div>
          )}
          <div className="description">
            <h2>DESCRIPTION</h2>
            <h3>
              {pokemon.description}
            </h3>
          </div>
        </div>
        <div className="stats-image">
          <img src={pokemon.hires} alt={pokemon.id} />
        </div>
        <div className="inner2 stats">
          <h2>POKEMON STATS</h2>
          <div>
            {completed.map((stat) => (
              <div
                key={stat.color}
              >
                <h2>{stat.name}</h2>
                <ProgressBar
                  bgcolor={stat.color}
                  completed={stat.stat}
                  tag={stat.color}
                />
              </div>
            ))}
          </div>
        </div>
      </div>
    </>

  );
};

PokemonStatCard.defaultProps = {
  pokemon: {
    id: 0,
    name: '',
    english: '',
    height: '',
    weight: '',
    type: [],
    evolution: {},
    ability: [],
    description: '',
    hires: '',
  },
};

PokemonStatCard.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    english: PropTypes.string,
    height: PropTypes.string,
    weight: PropTypes.string,
    type: PropTypes.arrayOf(PropTypes.string),
    evolution: PropTypes.shape({
      next: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
    }),
    ability: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
    description: PropTypes.string,
    hires: PropTypes.string,
    HP: PropTypes.number,
    Attack: PropTypes.number,
    Defense: PropTypes.number,
    SpecialAttack: PropTypes.number,
    SpecialDefense: PropTypes.number,
    Speed: PropTypes.number,
  }),
};

export default PokemonStatCard;
