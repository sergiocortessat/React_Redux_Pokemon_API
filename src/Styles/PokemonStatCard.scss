.wrapper {
  margin: 1em auto;
  display: inline-block;
  align-items: center;

  .inner {
    margin: 10px 50px 50px 10px;
    border-radius: 10px;
    border: 0.5px solid rgba(247, 85, 85, 0.6);
    padding: 10px;
    height: auto;
    background:
      linear-gradient(
        15deg,
        #dd580a 64%,
        #bd6328 70%,
        #d3561d 81%,
        rgb(240, 89, 70) 86%,
        #962f07 89%,
        #c2ab60 100%
      );

    * {
      margin-bottom: 8px;
    }

    .stat-type {
      display: flex;
      align-items: center;

      h3 {
        margin-left: 10px;
      }

      .type0 {
        background-color: red;
        padding: 3px;
        border: 0.5px solid gray;
        border-radius: 10px;
      }

      .type1 {
        background-color: blue;
        padding: 3px;
        border: 0.5px solid gray;
        border-radius: 10px;
      }
    }

    .stat-ability {
      display: flex;
      align-items: center;

      h3 {
        margin-left: 10px;
      }

      .ability1 {
        background-color: red;
        padding: 3px;
        border: 0.5px solid gray;
        border-radius: 10px;
      }

      .ability2 {
        background-color: blue;
        padding: 3px;
        border: 0.5px solid gray;
        border-radius: 10px;
      }
    }

    .stat-image {
      width: 100%;
      max-width: 80%;
    }

    .description {
      margin: 15px 0;
      text-align: center;
      border: 0.5px solid black;
      border-radius: 5px;
      box-sizing: border-box;
      background: linear-gradient(
        15deg,
        #cad5b5 64%,
        #dde2d4 70%,
        #dde2d4 81%,
        #fff 86%,
        #dde2d4 89%,
        #dde2d4 100%
      );
    }
  }

  .inner2 {
    margin: 40px 50px 50px 50px;
    text-align: center;
    height: 400px;

    h2 {
      margin-bottom: 10px;
    }

    > * div {
      margin-bottom: 10px;
    }
  }

  .inner2:hover {
    transform: rotate(0);
  }
}

.loading {
  text-align: center;
  margin-top: 15px;
  font-size: 40px;
}

@media screen and (min-width: 767px) {
  .wrapper {
    perspective: 1000px;
    // perspective-origin: 75% 70%;
    margin: 3em 1em;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    .inner {
      transition: 0.3s;
      transform: rotateY(40deg);
      width: 300px;
      height: auto;
    }

    .inner:hover {
      transform: rotate(0);
    }

    .inner2 {
      transition: 0.3s;
      transform: rotateY(-40deg);
      width: 300px;
      height: 400px;
    }

    .inner2:hover {
      transform: rotate(0);
    }
  }
}
