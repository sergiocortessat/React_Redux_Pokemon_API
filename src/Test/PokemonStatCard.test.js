import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import PokemonStatCard from '../App/Components/PokemonStatCard';

describe('PokemonStat Card with Link', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <PokemonStatCard />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
