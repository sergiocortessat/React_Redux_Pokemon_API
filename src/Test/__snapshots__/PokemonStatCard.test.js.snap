// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonStat Card with Link should match with snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="inner"
  >
    <div
      className="info"
    >
      <h2>
        ID
         
        #
      </h2>
      <h2>
        HEIGHT
         
        
      </h2>
      <h2>
        WEIGHT
         
        PokemonStats
        
      </h2>
      <div
        className="stat-type"
      >
        <h2>
          TYPE
        </h2>
        <h3
          className="type0"
        />
        <h3
          className="type1"
        />
      </div>
      <h2>
        NEXT-EVOLUTION
         
        NO-EVOLUTION
      </h2>
    </div>
    <div
      className="stat-ability"
    >
      <h2>
        ABILITY
      </h2>
      <h3
        className="ability1"
      >
        --
      </h3>
      <h3
        className="ability2"
      >
        --
      </h3>
    </div>
    <div
      className="description"
    >
      <h2>
        DESCRIPTION
      </h2>
      <h3>
        
      </h3>
    </div>
  </div>
  <div
    className="stats-image"
  >
    <img
      alt={Array []}
      src=""
    />
  </div>
  <div
    className="inner2 stats"
  >
    <h2>
      POKEMON STATS
    </h2>
    <div />
  </div>
</div>
`;
